crates/rethnet_rpc_server/src/node.rs

```diff
@@ -238,15 +291,16 @@ impl NodeData {
             None
         };
 
-        let block_gas_limit = *self.mem_pool.block_gas_limit();
         let result = mine_block(
             &mut *self.blockchain,
-            &mut *self.state,
+            self.state.clone(),
             &mut self.mem_pool,
             &self.evm_config,
             block_timestamp,
-            block_gas_limit,
             self.beneficiary,
+            self.min_gas_price,
+            // TODO: make this configurable
```

Comment:
Could you add an issue for this?

Reply:
Resolved in https://github.com/NomicFoundation/hardhat/pull/4254/commits/5f1e7d44273bf2bd16303c1873d9ccb71614bf1f